fastai/transforms.py

```diff
@@ -61,6 +61,26 @@ def center_crop(im, min_sz=None):
     start_c = math.ceil((c-min_sz)/2)
     return crop(im, start_r, start_c, min_sz)
 
+def googlenet_resize(im, targ, min_area_frac, min_aspect_ratio, max_aspect_ratio, flip_p):
+    h,w,*_ = im.shape
+    area = h*w
+    for _ in range(10):
+        targetArea = random.uniform(min_area_frac, 1.0) * area
+        aspectR = random.uniform(min_aspect_ratio, max_aspect_ratio)
+        ww = int(np.sqrt(targetArea * aspectR) + 0.5)
+        hh = int(np.sqrt(targetArea / aspectR) + 0.5)
+        if flip_p:
+            ww, hh = hh, ww
+        if hh <= h and ww <= w:
+            x1 = 0 if w == ww else random.randint(0, w - ww)
+            y1 = 0 if h == hh else random.randint(0, h - hh)
+            out = im[y1:y1 + hh, x1:x1 + ww]
+            out = cv2.resize(out, (targ, targ), interpolation=cv2.INTER_CUBIC)
```

Comment:
cv2 docs recommend INTER_AREA for making smaller. But for is_y=True transforms we need INTER_NEAREST if its COORD or CLASS. There's a partially working version in other transforms such as `Scale` you can borrow from, although I don't think they handle PIXEL correctly yet (should use INTER_AREA in that case)